tests/library/route-web-socket.spec.ts

```diff
@@ -508,3 +508,27 @@ test('should throw when connecting twice', async ({ page, server }) => {
   const error = await promise;
   expect(error.message).toContain('Already connected to the server');
 });
+
+test('should work with no trailing slash', async ({ page, server }) => {
+  const log: string[] = [];
+  // No trailing slash!
+  await page.routeWebSocket('ws://localhost:' + server.PORT, ws => {
+    ws.onMessage(message => {
+      log.push(message as string);
+      ws.send('response');
+    });
+  });
+
+  await page.goto('about:blank');
+  await page.evaluate(({ port }) => {
+    window.log = [];
+    // No trailing slash!
+    window.ws = new WebSocket('ws://localhost:' + port);
```

Comment:
I'd try to fix it the other way around, so that we would match `/` to `new WebSocket("ws://localhost:1234")` because on the actual networking layer the WebSocket has a trailing slash, it gets added by the browser. This should also require less hacks for the clients. Since having a WebSocket on `/` seems such a rare scenario, I'd prefer this over a lax matching logic. wdyt?

Looking at https://github.com/microsoft/playwright/blob/a395fb22c4cd414e58a291f2b1d87aa3a3714043/packages/playwright-core/src/server/injected/webSocketMock.ts#L153.